server/tserver/src/main/java/org/apache/accumulo/tserver/log/LogSorter.java

```diff
@@ -110,7 +109,13 @@ public void sort(String name, Path srcPath, String destPath) {
       String formerThreadName = Thread.currentThread().getName();
       int part = 0;
       try {
+        // check for finished first since another thread may have already done the sort
+        if (fs.exists(SortedLogState.getFinishedMarkerPath(destPath))) {
```

Comment:
Did you have specific reasons for putting the check here (and not for example at the beginning of the method?).  I noticed putting the check here will execute the finally block, which seems like a good thing.

Reply:
Mainly for the try/catch/finally since the `exists()` call has a checked IOException.  I wanted to add it earlier, in the process method before it gets added to the `currentWork` map but there isn't a try/catch there.  Then the more I thought about it, its probably not a bad thing it gets added to the `currentWork` map for this check since the `exists()` method is a Namenode operation.

Reply:
I think ideally the DistributedWorkQueue would be able to know that the sort was finished but that would be too much work for 1.10.  Maybe we could make improvements for 2.1

Reply:
Thanks for the explanation @milleruntime, it helped me understand the larger context.

>  Then the more I thought about it, its probably not a bad thing it gets added to the currentWork map for this check since the exists() method is a Namenode operation.

That is a nice benefit.

> I think ideally the DistributedWorkQueue would be able to know that the sort was finished but that would be too much work for 1.10. Maybe we could make improvements for 2.1

I have not looked at the DistributedWorkQueue in a while.  If it will always have race conditions and possibly run things multiple times, then your approach here of doing this check as late as possible will be hard to beat.